### PR TITLE
[2.11.x] DDF-3589 Use the original encoded SAML request when validating it after login

### PR DESCRIPTION
**Pull request by blen-desta**
#### What does this PR do?
Use the original encoded SAML request when validating it after login

#### Who is reviewing it? 
@Lambeaux @brjeter @bakejeyner 

#### Choose 2 committers to review/merge the PR. 
@coyotesqrl
@rzwiefel
@stustison

#### How should this be tested? (List steps with links to updated documentation)
Connect with Spring SP

#### Any background context you want to provide?
SAML SPEC (https://www.oasis-open.org/committees/download.php/35387/sstc-saml-bindings-errata-2.0-wd-05-diff.pdf)
```
Further, note that URL-encoding is not canonical; that is, there are multiple legal 
encodings for a given value. The relying party MUST therefore perform the verification 
step using the original URL-encoded values it received on the query string. It is not 
sufficient to re-encode the parameters after they have been processed by software 
because the resulting encoding  may not match the signer's encoding.
```
#### What are the relevant tickets?
[DDF-3589](https://codice.atlassian.net/browse/DDF-3589)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java [hunk 1]
```diff
@@ -601,13 +601,9 @@ private Response showLoginPage(
...
       } else {
         LOGGER.debug("Building the JSON map to embed in the index.html page for login.");
-        Document doc = DOMUtils.createDocument();
-        doc.appendChild(doc.createElement("root"));
-        String authn = DOM2Writer.nodeToString(OpenSAMLUtil.toDom(authnRequest, doc, false));
-        String encodedAuthn = RestSecurity.deflateAndBase64Encode(authn);
         responseMap.put(PKI, hasCerts);
         responseMap.put(GUEST, guestAccess);
-        responseMap.put(SAML_REQ, encodedAuthn);
+        responseMap.put(SAML_REQ, samlRequest);
```
Comment (rzwiefel) [inline platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java]: Do you think we should leave a inline comment that the *original* samlRequest must be used as decoding/re-encoding will cause issues? Dunno if we will ever have anyone try to refactor code in here but it'd be a shame to have them make that mistake.

Comment (blen-desta) [reply platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java]: That's a good idea. I'll make that change. I'll add the spec too.

**Comment by blen-desta** [comment]
@brjeter Only one and I changed the request encoding to work with this change.

**Comment by blen-desta** [comment]
build now

**Pull request merged**